lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -205,6 +205,203 @@ sub schedules {
+                    if (!defined $machine_name) {
+                        $machine_name = $yaml->{$group}->{defaults}->{$arch}{machine};
+                    }
+                    my $machine = $self->schema->resultset("Machines")->find({name => $machine_name});
+                    my $product = $self->schema->resultset("Products")->find(
+                        {
+                            arch    => $arch,
+                            distri  => $yaml->{$group}->{products}{$product_name}{distribution},
+                            flavor  => $yaml->{$group}->{products}{$product_name}{flavor},
+                            version => $yaml->{$group}->{products}{$product_name}{version},
+                        });
+                    my $test_suite
+                      = $self->schema->resultset("TestSuites")->find_or_create({name => $testsuite_name});
+
+                    if (!$machine || !$product) {
+                        push @$errors, "Machine '$machine_name' invalid" unless $machine;
+                        push @$errors, "Product '$product_name' invalid" unless $product;
+                        last;
+                    }
+
+                    my $r = $self->schema->resultset("JobTemplates")->find_or_create(
+                        {
+                            product_id    => $product->id,
+                            machine_id    => $machine->id,
+                            group_id      => $id,
+                            test_suite_id => $test_suite->id,
+                        });
+
+                    my $values = {
+                        product_id => $product->id,
+                        machine_id => $machine->id,
+                    };
+                    if ($prio) {
+                        $values->{prio} = $prio;
+                    }
+                    eval { $r->update($values); };
+                    push @$errors, $@ if $@;
+
+                    # Stop iterating if there were errors with this test suite
+                    last if (@$errors);
+                }
+            }
+        }
+
+        if (@$errors) {
+            $json->{error} = \@$errors;
+            $self->app->log->error(@$errors);
+            $self->respond_to(
+                json => {json => $json, status => 400},
+                html => sub {
+                    $self->flash('error', "Error applying job template: \@errors");
+                    $self->res->code(303);
+                    $self->redirect_to($self->req->headers->referrer);
+                },
+            );
+            return;
+        }
+
+        $self->emit_event('openqa_jobtemplate_create', $json);
+        $self->respond_to(
+            json => {json => $json},
+            html => sub {
+                $self->flash(info => 'Job template added');
+                $self->res->code(303);
+                $self->redirect_to($self->req->headers->referrer);
+            });
+
+        # Only one group will be processed
```

Comment:
I prefer imperative for comments (similar to commit messages) if it makes sense, e.g. `# Process only one group`. That would also be consistent with the rest of your comments :-)